### PR TITLE
Improve API of LIteralData to provide more flexibility

### PR DESCRIPTION
**Pull request by DoumanAsh**

### CODE REVIEW

File: src/packet/literal_data.rs [hunk 1]
```diff
@@ -84,6 +84,24 @@ impl LiteralData {
...
 
+    #[inline]
+    /// Extracts data in to raw data
+    pub fn into_bytes(self) -> Vec<u8> {
+        self.data
+    }
+
+    #[inline]
+    /// Extracts data as string, returning raw bytes as Err if not valid utf-8 string
+    pub fn into_string(self) -> Result<String, Vec<u8>> {
```
Comment (dignifiedquire) [inline src/packet/literal_data.rs:95]: Given that `to_string` returns `None` in the error case, I think this api should mirror that. If you want this exact behaviour, I would suggest renaming this method into sth like `try_into_string`

Comment (DoumanAsh) [reply src/packet/literal_data.rs:95]: This behavior is not very useful for owned version because there is no way to examine what is wrong with data

Comment (hko-s) [reply src/packet/literal_data.rs:95]: Renaming to `pub fn try_into_string(self) -> Result<String, Vec<u8>>` seems like a good approach to me, then

Comment (hko-s) [reply src/packet/literal_data.rs]: From the point of view of OpenPGP semantics, I tend to think that `DataMode::Binary` should also be handled like the `_` case (i.e.: I think the `match` should be dropped).

As far as I can tell, the `DataMode` of a literal packet isn't very intimately linked with the expected content of the packet. So if we're willing to try `String::from_utf8` for `DataMode::Binary`, I think we should do for all cases.

File: src/packet/literal_data.rs [hunk 2]
```diff
@@ -84,6 +84,24 @@ impl LiteralData {
...
+    /// Extracts data in to raw data
+    pub fn into_bytes(self) -> Vec<u8> {
+        self.data
+    }
+
+    #[inline]
+    /// Extracts data as string, returning raw bytes as Err if not valid utf-8 string
+    pub fn try_into_string(self) -> Result<String, Vec<u8>> {
+        match self.mode {
+            DataMode::Binary => Err(self.data),
```
Comment (hko-s) [inline src/packet/literal_data.rs:97]: From the point of view of OpenPGP semantics, I tend to think that DataMode::Binary should also be handled like the _ case (i.e.: I think the match should be dropped).

As far as I can tell, the DataMode of a literal packet isn't very intimately linked with the expected content of the packet. So if we're willing to try String::from_utf8 for DataMode::Binary, I think we should do for all cases.

Comment (DoumanAsh) [reply src/packet/literal_data.rs:97]: I followed `to_string` implementation which ignores Binary variant 🤔

Comment (hko-s) [reply src/packet/literal_data.rs:97]: Hm, I guess that's fair enough. Sorry, ignore this point :)

**Review by dignifiedquire — APPROVED** [review]
thanks

**Pull request merged**